docs/manual/generics.tex

```diff
@@ -853,6 +856,95 @@
   void m(@Nullable Object obj)
 \end{Verbatim}
 
+\sectionAndLabel{Classes with implicit qualifier parameters}{class-qualifier-polymorphism}
+
+The Checker Framework supports type \emph{qualifier} polymorphism for classes.  This is similar to
+Java's generic classes, but is used in situations where you cannot use Java generics.
+
+To add a qualifier parameter to a class, annotate its declaration with \refqualclass{framework/qual}{HasQualifierParameter}.
+
+\begin{Verbatim}
+@HasQualifierParameter(Tainted.class)
+class StringBuffer { ... }
+\end{Verbatim}
+
+\<@HasQualifierParameter> analogizes \<StringBuffer> to a
+generic collection.  A qualifier on a use of \<StringBuffer> is treated as
+appearing both on the \<StringBuffer> and on its conceptual type argument.  That
+is:\\
+\begin{tt}
+  @Tainted StringBuffer $\approx$ @Tainted Collection<@Tainted Character>\\
+  @Untainted StringBuffer $\approx$ @Untainted Collection<@Untainted Character>
+\end{tt}
+
+If two types have different qualifier arguments, they have no subtyping
+relationship.  (This is ``invariant subtyping'', also used by Java for
+generic classes.)  In particular, \<@Untainted StringBuffer> is not a
+subtype of \<@Tainted StringBuffer>; an attepmt to cast between them (in
+either direction) will yield an \<invariant.cast.unsafe> error.
+
+If a class extends a \<@HasQualifierParameter> class (or implements a
+\<@HasQualifierParameter> interface), then that class must also be marked
+\<@HasQualifierParameter>.
+
+\subsectionAndLabel{Using class qualifier parameters in method signatures}{class-qual-param-method}
+
+In a class with a class qualifier parameter, a formal parameter type should
+sometimes have the same qualifier as the class qualifier parameter.  To express
+this, annotate both the receiver type and the parameter type with the
```

Comment:
It seems that `@PolyX` in a method can have two different meanings:
 * method qualifier polymorphism
 * class implicit qualifier polymorphism

How does a programmer know what it means, and how does a programmer choose one of those meanings?
(I suspect that if the class has implicit qualifier polymorphism, then the meaning is always the latter, and no method qualifier polymorphism is possible.)
There should also be a mention of this and a cross-reference from the method qualifier polymorphism section.

Reply:
I addressed this in the paragraph above this one.